ArrayUtilsLib/truffle/test/TestArrayUtilsLib.sol

```diff
@@ -180,4 +181,33 @@ contract TestArrayUtilsLib{
     Assert.equal(resultArray, expectedArray, "heapSort");
   }
 
+  function testUniqFunction(){
+    delete expectedArray;
+
+    expectedArray.push(1);
+    expectedArray.push(2);
+    expectedArray.push(7);
+    expectedArray.push(4);
+    expectedArray.push(0xff3);
+    expectedArray.push(0);
+    expectedArray.push(1095);
+
+    /*
+      Currently delete array[index] does not actually delete the element, it
+      only replaces the item defined in the index with a zero. So the uniq
+      function will push any trailing zeroes to the end of the array and update
+      the array length to avoid any mistake while iterating over array.length
+      Discussion: https://github.com/ethereum/solidity/issues/414
+    */
+    expectedArray.push(0);
+    expectedArray.push(0);
+    expectedArray.push(0);
```

Comment:
I'm still not fully convinced about this. Even after decrementing the length of array on Array256Lib.sol (line 219), the result array still has length 10 instead of 7, with 3 trailing zeroes at the end that replaced the duplicates.

There must be a solution/alternative for this.

Reply:
Also check: https://github.com/ethereum/solidity/issues/414